torch_geometric/explainability/explainer.py

```diff
@@ -0,0 +1,221 @@
+from dataclasses import dataclass
+from typing import Callable, Tuple
+
+import torch
+
+from torch_geometric.explainability.algo.base import ExplainerAlgorithm
+from torch_geometric.explainability.algo.captumexplainer import CaptumExplainer
+from torch_geometric.explainability.explanations import Explanation
+from torch_geometric.explainability.utils import to_captum
+
+
+@dataclass
+class Threshold:
+    type: str
+    value: float
+    _valid_type = ["soft", "hard", "connected"]
```

Comment:
"soft" means that we don't threshold right?

should we also distinguish hard by "top K" or an absolute value?

Reply:
Yes you're totally right! `TopK` sounds good